sw/device/silicon_creator/rom/e2e/weak_straps/gpio.bzl

```diff
@@ -43,33 +43,29 @@ def strap_combination(strap):
             settings["b{}_value".format(i)] = "true"
     return settings
 
-def strap_combination_test(name, rom, value, evaluator = None, tags = [], extra_verilator_args = []):
+def strap_combination_test(name, rom, value, evaluator = None, tags = [], extra_verilator_args = ""):
     settings = strap_combination(value)
     if evaluator == None:
-        evaluator = r"console --exit-success=\"{pass}\" --exit-failure=\"{fail}\""
-    evaluator = evaluator.format(**settings)
+        evaluator = "console --exit-success=\"{pass}\" --exit-failure=\"{fail}\""
+    settings["evaluator"] = evaluator.format(**settings)
 
-    opentitan_functest(
+    opentitan_test(
         name = name,
-        srcs = ["sw_straps_test.c"],
-        signed = False,
         tags = tags,
-        targets = [
-            "verilator",
-        ],
-        ot_flash_binary = "//sw/device/silicon_creator/rom/e2e:empty_test_slot_a",
+        exec_env = {"//hw/top_earlgrey:sim_verilator": None},
         verilator = verilator_params(
             rom = rom,
-            test_cmds = extra_verilator_args + [
-                "--exec=\"gpio set IOC0 --mode={b0_mode} --value={b0_value} --pull={b0_pull}\""
-                    .format(**settings),
-                "--exec=\"gpio set IOC1 --mode={b1_mode} --value={b1_value} --pull={b1_pull}\""
-                    .format(**settings),
-                "--exec=\"gpio set IOC2 --mode={b2_mode} --value={b2_value} --pull={b2_pull}\""
-                    .format(**settings),
-                "--exec=\"{}\"".format(evaluator),
-                "no-op",
-            ],
+            assemble = "{firmware}@0x0",
```

Comment:
The `assemble` param has no effect in verilator tests.  Currently the test_dispatch function for sim_dv and sim_verilator cannot assemble a multi-stage image.

Reply:
Fixed.